SQL/Archive/2018-02-15_AddCustomizableCRFsUnderLinksInImagBrowser.sql

```diff
@@ -0,0 +1,33 @@
+-- Add Imaging Browser to Imaging Modules
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'ImagingBrowserLinkedInstruments', 'Instruments that the users want to see linked from Imaging Browser', 1, 1, 'instrument', ID, 'Imaging Browser Links to Instruments', 7 FROM ConfigSettings WHERE Name="imaging_modules";
+
+-- Default imaging_browser settings for Linked instruments
+-- This will be the two tables mri_parameter_form and radiology_review; IF they exist in the database
+SET @s1 = (SELECT IF(
+    (SELECT COUNT(*)
+        FROM INFORMATION_SCHEMA.TABLES
+        WHERE table_name = 'mri_parameter_form'
+        AND table_schema = DATABASE()
+    ) > 0,
+    "INSERT INTO Config (ConfigID, Value) SELECT cs.ID, 'mri_parameter_form' FROM ConfigSettings cs WHERE cs.Name='ImagingBrowserLinkedInstruments'",
+    "SELECT 'No' as 'mri parameter_form table exists'"
+));
```

Comment:
recommendation by @xlecours following his SNP patch `Archive/18.0/2017-02-28_SNP_table_structure_check.sql`.

@PapillonMcGill 
any better way to conditionally insert?

Reply:
The way to conditionally enter values in Config table if others tables exist is effectively to query INFORMATION_SCHEMA. 
On the usage of server side prepared statement, it is usually done to save time on repeated query, not exactly the case here as 2 prepared statements are used. Also, there is no real performance issues here. 
In this case, I would use the variable to hold the table names ('mri_parameter_form' or 'radiology_review') and create the prepared statement from the instruction like on https://dev.mysql.com/doc/refman/5.7/en/sql-syntax-prepared-statements.html and execute the statement twice (once with each value).

Although statements are cleared right after, there is a limitation on the number of concurrent prepared statements in a database, so I would limit the usage to only where it is critical to use.

Also, variables are not manually cleared and with last for the duration of the database connection.

Would it be easy to put the SQL code in an 'upgrade' PHP function?